deploy/docker/cp-edge/sync-routes.py

```diff
@@ -500,6 +515,133 @@ def load_pods_for_runs_with_endpoints():
                                 pods_with_endpoints.append(pod)
         return pods_with_endpoints
 
+
+def create_dns_record(service_spec):
+        if hosted_zone_base_value is not None:
+                dns_custom_domain = service_spec["edge_location"] + "." + hosted_zone_base_value
+                dns_record_create = os.path.join(api_url, API_POST_DNS_RECORD
+                                                 + "?regionId={regionId}"
+                                                 .format(regionId=service_spec["cloudRegionId"]))
+                data = json.dumps({
+                        'dnsRecord': dns_custom_domain,
+                        'target': "{external_ip}".format(external_ip=edge_service_external_ip)
+                })
+                dns_record_create_response = call_api(dns_record_create, data)
+                if dns_record_create_response and "payload" in dns_record_create_response and "status" in \
+                        dns_record_create_response["payload"] \
+                        and dns_record_create_response["payload"]["status"] == "INSYNC":
+                        service_spec["custom_domain"] = dns_custom_domain
+                        service_spec["edge_location"] = None
+                else:
+                        log_task_event("CreateDNSRecord",
+                                       "Fail to create DNS record for the run",
+                                       service_spec["run_id"],
+                                       service_spec["pod_id"],
+                                       "FAILURE")
+                        update_run_status(service_spec["run_id"], "FAILURE")
+                        raise ValueError("Couldn't create DNS record")
```

Comment:
I would propose not to throw any exceptions without handling them explicitly. I'm not sure enough how unhanded exceptions will affect threads in the pool. In order to avoid the worst scenario when a single failing endpoint prevents other endpoints from being processed I suggest to handle errors explicitly.

Of course If you've tested it and you are pretty confident that it is safe then I'm good too.

Reply:
Good point.
It should work fine, exception will be raised in the thread and shouldn't affect other threads, but I think it is better to explicitly catch it and just message log about it and quit from thread